examples/sugarscape_g1mt/sugarscape_g1mt/trader_agents.py

```diff
@@ -0,0 +1,352 @@
+        mrs_other = other.calculate_MRS()
+
+        # calculate each agents welfare
+        welfare_self = self.calculate_welfare(self.sugar, self.spice)
+        welfare_other = other.calculate_welfare(other.sugar, other.spice)
+
+        if math.isclose(mrs_self, mrs_other):
+            return
+
+        # calcualte price
+        price = math.sqrt(mrs_self * mrs_other)
+
+        if mrs_self > mrs_other:
+            # self is a sugar buyer, spice seller
+            sold = self.maybe_sell_spice(other, price, welfare_self, welfare_other)
+            # no trade - criteria not met
+            if not sold:
+                return
+        else:
+            # self is a spice buyer, sugar seller
+            sold = other.maybe_sell_spice(self, price, welfare_other, welfare_self)
+            # no trade - criteria not met
+            if not sold:
+                return
+
+        # Capture data
+        self.prices.append(price)
+        self.trade_partners.append(other.unique_id)
+
+        # continue trading
+        self.trade(other)
+
+    ######################################################################
+    #                                                                    #
+    #                      MAIN TRADE FUNCTIONS                          #
+    #                                                                    #
+    ######################################################################
+
+    def move(self):
+        '''
+        Function for trader agent to identify optimal move for each step in 4 parts
+        1 - identify all possible moves
+        2 - determine which move maximizes welfare
+        3 - find closest best option
+        4 - move
+        '''
+
+        # 1. identify all possible moves
+
+        neighbors = [i
+                     for i in self.model.grid.get_neighborhood(
+                self.pos, self.moore, True, self.vision
+            ) if not self.is_occupied_by_other(i)]
+
+        # 2. determine which move maximizes welfare
+
+        welfares = [
+            self.calculate_welfare(
+                self.sugar + self.get_sugar_amount(pos),
+                self.spice + self.get_spice_amount(pos))
+            for pos in neighbors
+        ]
+
+        # 3. Find closest best option
+
+        # find the highest welfare in welfares
+        max_welfare = max(welfares)
+        # get the index of max welfare cells
+        candidate_indices = [i for i in range(len(welfares))
+                             if math.isclose(welfares[i], max_welfare)]
+
+        # convert index to positions of those cells
+        candidates = [neighbors[i] for i in candidate_indices]
+
+        min_dist = min(get_distance(self.pos, pos) for pos in candidates)
+
+        final_candidates = [pos for pos in candidates
+                            if math.isclose(get_distance(self.pos, pos), min_dist, rel_tol=1e-02
+                                            )]
+        self.random.shuffle(final_candidates)
```

Comment:
Reposting https://github.com/projectmesa/mesa/pull/1290#discussion_r1060319796. Not critical, but adding the comment here just in case.
```python
final_candidate = self.random.choice(final_candidates)
self.model.grid.move_agent(self, final_candidate)
```
is clearer.